clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -109,6 +112,9 @@
     /* all the state related to transactions, in particular the PID, epoch, and sequence numbers */
     private final TransactionState transactionState;
 
+    /* A blocking version of the network client to be used for the new blocking RPCs for the idempotent producer */
+    private final BlockingNetworkClient blockingNetworkClient;
```

Comment:
It seems a little confusing to have two references to the same underlying NetworkClient. Maybe we should make this `NetworkClientUtils` or something like that and just implement the methods as static utility functions? Unfortunately java doesn't have great mixin patterns.